src/components/cluster/detail/scale_node_pool_modal.js

```diff
@@ -0,0 +1,439 @@
+  };
+
+  buttonProperties = () => {
+    var workerDelta = this.workerDelta();
+    var pluralizeWorkers = this.pluralize(workerDelta);
+
+    if (
+      this.supportsAutoscaling(
+        this.props.provider,
+        this.props.cluster.release_version
+      )
+    ) {
+      if (this.state.scaling.min > this.props.workerNodesDesired) {
+        workerDelta = this.state.scaling.min - this.props.workerNodesDesired;
+        return {
+          title: `Increase minimum number of nodes by ${workerDelta}`,
+          style: 'success',
+          disabled: !this.state.scaling.minValid,
+        };
+      }
+
+      if (this.state.scaling.max < this.props.workerNodesDesired) {
+        workerDelta = Math.abs(
+          this.props.workerNodesDesired - this.state.scaling.max
+        );
+        return {
+          title: `Remove ${workerDelta} worker node${this.pluralize(
+            workerDelta
+          )}`,
+          style: 'danger',
+          disabled: !this.state.scaling.maxValid,
+        };
+      }
+
+      if (this.state.scaling.min != this.props.cluster.scaling.min) {
+        return {
+          title: 'Apply',
+          style: 'success',
+          disabled: !(
+            this.state.scaling.minValid && this.state.scaling.maxValid
+          ),
+        };
+      }
+
+      if (this.state.scaling.max != this.props.cluster.scaling.max) {
+        return {
+          title: 'Apply',
+          style: 'success',
+          disabled: !(
+            this.state.scaling.minValid && this.state.scaling.maxValid
+          ),
+        };
+      }
+
+      return {
+        disabled: true,
+      };
+    }
+
+    if (workerDelta === 0) {
+      return {
+        disabled: true,
+      };
+    }
+
+    if (workerDelta > 0) {
+      return {
+        title: `Add ${workerDelta} worker node${pluralizeWorkers}`,
+        style: 'success',
+        disabled: !(this.state.scaling.minValid && this.state.scaling.maxValid),
+      };
+    }
+
+    if (workerDelta < 0) {
+      return {
+        title: `Remove ${Math.abs(workerDelta)} worker node${pluralizeWorkers}`,
+        style: 'danger',
+        disabled: !(this.state.scaling.minValid && this.state.scaling.maxValid),
+      };
+    }
+
+    return {
+      disabled: true,
+    };
+  };
+
+  render() {
+    var warnings = [];
+    if (
+      this.state.scaling.max < this.props.workerNodesRunning &&
+      this.state.scaling.minValid
+    ) {
+      var diff = this.props.workerNodesRunning - this.state.scaling.max;
+
+      if (
+        this.supportsAutoscaling(
+          this.props.provider,
+          this.props.cluster.release_version
+        )
+      ) {
+        warnings.push(
+          <CSSTransition
+            classNames='rollup'
+            enter={true}
+            exit={true}
+            key={1}
+            timeout={this.rollupAnimationDuration}
+          >
+            <p key='node-removal'>
+              <i className='fa fa-warning' /> The cluster currently has{' '}
```

Comment:
Would instances of 'cluster' have to be changed to 'nodepool' in the text here?

Reply:
Changes committed and pushed

Yeah there are more changes to do here, I just wanted to split them into various PRs in order to make PRs more reviewable. I stated that in my first comment, sorry if I wasn't clear enough

> This PR is just for:
> 
> * Triggering the modal, and
> * Updating values in the store.
>